public/components/datasources/components/manage/data_connection.tsx

```diff
@@ -294,6 +344,55 @@ export const DataConnection = (props: any) => {
     );
   };
 
+  useEffect(() => {
+    // Loads databases cache when cache is empty
+    if (datasourceDetails.name) {
+      const cache = CatalogCacheManager.getDataSourceCache().dataSources.find(
+        (datasource) => datasource.name === datasourceDetails.name
+      );
+      if (cache?.status !== 'Updated') {
+        startLoadingDatabases(datasourceDetails.name);
+        setIsRefreshing(true);
+      } else {
+        const cachedList: CachedDatabase[] =
+          CatalogCacheManager.getDataSourceCache().dataSources.find(
+            (cachedDataSource) => cachedDataSource.name === datasourceDetails.name
+          )?.databases || [];
+        if (cachedList) {
```

Comment:
Question: since online 360 you are doing `... || []` then this line it will always be evaluated as truthy is it? then do you want to check if length > 0 or remove this if to always set the state?